Sockets/BackEndMonitoringSockets/src/CSocketWrapper.cpp

```diff
@@ -43,25 +41,26 @@ std::string CSocketWrapper::Receive(const int socket)
 
 bool CSocketWrapper::Send(const int socket, const std::string& line)
 {
-	if (!SendMessageLength(socket, line.length()))
+	int line_length = static_cast<int>(line.length());
+	if (!SendMessageLength(socket, line_length))
 	{
 		return false;
 	}
 
-	if (send(socket, line.c_str(), line.length(), 0) == CONNECTION_ERROR)
+	if (send(socket, line.c_str(), line_length, 0) == CONNECTION_ERROR)
 	{
 		return false;
 	}
 	return true;
 }
 
-bool CSocketWrapper::IsAllDataReceived(size_t msg_size,
-	size_t received_msg_size) const
+bool CSocketWrapper::IsAllDataReceived(int msg_size,
+	int received_msg_size) const
 {
 	return msg_size == received_msg_size;
 }
 
-bool CSocketWrapper::SendMessageLength(const int socket,  size_t length)
+bool CSocketWrapper::SendMessageLength(const int socket,  int length)
 {
 	if (send(socket, reinterpret_cast<char*>(&length),
```

Comment:
Check send and recv type, and use this return value to check if everything is ok